src/interface/viewport.c

```diff
@@ -2632,13 +2634,321 @@ void store_interaction_info(paint_struct *ps)
 	}
 }
 
+static bool sub_679236_679662_679B0D_679FF1(uint32 ebx, rct_g1_element *image, uint8 *esi) {
+	if (!(image->flags & G1_FLAG_BMP)) {
+		assert(false);
```

Comment:
Ah yes this is the bmp function but also the corrupt element function. Suggest taking this check out of here and putting it into the main function.